x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/aggregate/Max.java

```diff
@@ -39,4 +40,12 @@ public DataType dataType() {
     public String innerName() {
         return "max";
     }
+
+    @Override
+    protected TypeResolution resolveType() {
+        return Expressions.typeMustBe(
+            field(),
+            e -> e.dataType().isNumeric() || e.dataType() == DataType.DATE,
+            Expressions.getErrorMessageForNumericRequirement(field()));
```

Comment:
The error message though is incorrect since it will indicate the type needs to be numeric but it can also be a date.
It might make sense to add a dedicated method in `Expressions.typeMustBeNumericOrDate` since it will come handy in the future with date arithmetic (date + int).